scripts/lib/spreadsheet-standardizer.ts

```diff
@@ -1,53 +1,92 @@
+  'amount.minimum',
+  'amount.maximum',
+  'amount.representative',
+];
 
 // Standardize column names or values in an incentive spreadsheet.
 export class SpreadsheetStandardizer {
-  private fieldMap: { [index: string]: string };
+  private fieldMap: FlatAliasMap;
+  private valueMap: { [index: string]: FlatAliasMap };
   private strict: boolean;
   private lowIncomeThresholds: LowIncomeThresholdsMap | null;
 
   /**
-   * @param fieldMap: a map from canonical name to all of the aliases it might appear as in the Google spreadsheets
+   * @param fieldMap: a map from canonical column name to all of the aliases it might appear as in the Google spreadsheets
+   * @param valueMap: a map from canonical value name to all of the aliases it might appear as in the Google spreadsheets
    * @param strict: whether to throw an error on aliases that aren't found
    * @param lowIncomeThresholds: state-segmented income thresholds. If empty, the script won't try to associate records with low-income programs.
    */
   constructor(
-    fieldMap: { [index: string]: string[] } = FIELD_MAPPINGS,
+    fieldMap: AliasMap = FIELD_MAPPINGS,
+    valueMap: { [index: string]: AliasMap } = VALUE_MAPPINGS,
     strict: boolean = true,
     lowIncomeThresholds: LowIncomeThresholdsMap | null = null,
   ) {
     // We reverse the input map for easier runtime use, since
     // we need to go from alias back to canonical name.
     this.fieldMap = reverseMap(fieldMap);
+    this.valueMap = reverseValueMap(valueMap);
     this.strict = strict;
     this.lowIncomeThresholds = lowIncomeThresholds;
   }
 
+  convertToCanonical(val: string, colName: string): string {
+    const components: string[] = ARRAY_FIELDS.includes(colName)
+      ? val.split(',').map(x => x.trim())
+      : [val];
+
+    return components
+      .map((component: string) => {
+        if (
+          this.valueMap[colName] !== undefined &&
+          this.valueMap[colName][cleanFieldName(component)] !== undefined
+        ) {
+          component = this.valueMap[colName][cleanFieldName(component)];
+        }
+        return component;
+      })
+      .join(',');
+  }
+
   // Convert a row with possible column aliases to a canonical column name.
   convertFieldNames(input: Record<string, string>): Record<string, string> {
     const output: Record<string, string> = {};
     for (const key in input) {
       const cleaned = cleanFieldName(key);
-      if (this.fieldMap[cleaned] === undefined) {
-        if (this.strict) {
-          throw new Error(`Invalid column found: ${cleaned}; original: ${key}`);
-        }
-        output[key] = input[key];
-        continue;
+      if (this.fieldMap[cleaned] === undefined && this.strict) {
+        throw new Error(`Invalid column found: ${cleaned}; original: ${key}`);
+      }
+
+      const newCol = this.fieldMap[cleaned] || key;
+      let val = this.convertToCanonical(input[key], newCol);
+
+      // Special case cleaning fields – eventually this can be
```

Comment:
maybe say more here about how this could improve?

It almost feels like a TODO - is there a format to use for that here?

Reply:
I'm realizing what I really meant by "more principled" is that it's feels a bit awkward as part of that method, so I extracted it to a separate method.